jupyterhub/utils.py

```diff
@@ -296,6 +299,30 @@ def metrics_authentication(self):
         raise web.HTTPError(403)
 
 
+def needs_scope(scope):
```

Comment:
Lots of direct unittests for this decorator, e.g.


```

class Test:

    @needs_scope("read:user")
    def foo(self, name):
        pass

obj = Test()
obj.scopes = []
with pytest.raises(web.HTTPError):
    obj.foo()
```

in a new `test_scopes.py`